src/Git.hs

```diff
@@ -141,6 +142,11 @@ data SomeRefSpec = forall a . RefSpec a => AsRefSpec a
 instance RefSpec SomeRefSpec where
   refSpec (AsRefSpec a) = refSpec a
 
+data ForceRefSpec = forall a . RefSpec a => ForceRefSpec a
+
+instance RefSpec ForceRefSpec where
+  refSpec (ForceRefSpec a) = "+" ++ refSpec a
```

Comment:
This doesn't quite align with the specification of ref-specs, which can have at most one optional `+`, but this construction would also allow you to build a `ForceRefSpec (ForceRefSpec ...)`.

Perhaps, the force-ness of the push should rather be a parameter of `AsRefSpec`?